tasks/init.rb

```diff
@@ -0,0 +1,62 @@
+#!/opt/puppetlabs/puppet/bin/ruby
+require 'facter'
+require 'json'
+
+params   = JSON.parse(STDIN.read)
+timeout  = params['timeout'].to_i
+message = params['messsage'] || nil
```

Comment:
nit: I think the `|| nil` isn't required.  If the `message` element is missing it will return nil anyway.